index.js

```diff
@@ -1,121 +1,177 @@
-			if(matches.length > 0){
-				title = matches[1].trim()
-			}
-		}catch(err){
-			title = $content('.box-info-heading').text();
-		}
-
-		info.title = title
-
-		$info_left = cheerio.load($content('ul.list').eq(1).html());
-		$info_right = cheerio.load($content('ul.list').eq(2).html());
-
-		
-		info.category = $info_left('li').eq(0).children('span').text().trim();
-		info.type = $info_left('li').eq(1).children('span').text().trim();
-		info.language = $info_left('li').eq(2).children('span').text().trim();
-		info.size = $info_left('li').eq(3).children('span').text().trim();
-		info.uploaded = $info_left('li').eq(4).children('span').text().trim();
-
-	
-		info.downloads = $info_right('li').eq(0).children('span').text().trim();
-		info.last_check = $info_right('li').eq(1).children('span').text().trim();
-		info.date_uploaded = $info_right('li').eq(2).children('span').text().trim();
-		info.seeders = $info_right('li').eq(3).children('span').text().trim();
-		info.leechers = $info_right('li').eq(4).children('span').text().trim();
-
-		info.download ={
-            magnet:$content('ul > li > a[href^=magnet]').eq(0).attr('href')
-		}
-
-
-		info.files = [];
-		$content('.file-content ul').each(function (i,el) {
-			info.files.push($content('.file-content ul').eq(i).children('li').text());
-		})
-
-	 return info || null;
-	});
+const search = opt => {
+  opt.category = validCategories.has(opt.category) ? opt.category : null;
+  opt.orderBy = validOrderBy.has(opt.orderBy) ? opt.orderBy : 'seeders';
+  opt.sortBy = validSortBy.has(opt.sortBy) ? opt.sortBy : 'desc';
+
+  let reqUrl = `${opt.url || url}/sort-${
+    opt.category ? 'category-' : ''
+  }search/${encodeURIComponent(opt.query)}/${
+    opt.category ? opt.category + '/' : ''
+  }${opt.orderBy}/${opt.sortBy}/${opt.page || 1}/`;
+
+  return got(reqUrl).then(data => {
+    let $ = cheerio.load(data.body);
+
+    let table = $('tbody > tr');
+
+    let torrents = [];
+
+    table.each((i, elem) => {
+      let chunk = cheerio.load(elem);
+
+      torrents.push({
+        title: chunk('.coll-1').text(),
+        href: chunk('a')
+          .eq(1)
+          .attr('href'),
+        seed: chunk('.coll-2').text(),
+        leech: chunk('.coll-3').text(),
+        size: chunk('.coll-4')
+          .children()
+          .remove()
+          .end()
+          .text(),
+        uploader: chunk('.coll-5').text(),
+      });
+    });
+
+    return {domain: opt.url || url, query: opt.query, torrents};
+  });
+};
+
+/**
+ * @method info
+ * @desc xTorrent info method
+ * @param {string} url - example http://1337x.org/torrent/738327/New-Girl-S03E14-HDTV-x264-LOL/
+ * @returns {function} promise
+ */
 
+const info = url => {
+  return got(url).then(data => {
+    let $detail = cheerio.load(data.body);
+    let $content = cheerio.load($detail.html());
+
+    let info = {};
+
+    info.title = $detail('title').text();
+    info.title = info.title.slice(9).substring(0, info.title.length - 25);
```

Comment:
still not safe, but instead use replace with regex example ```x.replace(/(Donwloand|Torrent)/, '')``` etc.

Reply:
I think it's safe since the site used the same pattern for all the titles and above mentioned replace regex might remove the words from the title from a valid torrent which uses those words.

Reply:
ok